pipeline/mutate/mutator_hydrator.go

```diff
@@ -254,6 +275,40 @@ func (a *MutatorHydrator) Config(config json.RawMessage) (*MutatorHydratorConfig
 		return nil, NewErrMutatorMisconfigured(a, err)
 	}
 
+	var rt http.RoundTripper
+
+	if c.PreAuth != nil && c.PreAuth.Enabled {
```

Comment:
Is this copy/pasted from another handler? If so, I think it would make sense to extract that logic into a common module which then is imported and used here :)

Reply:
It highly influenced by the client credential from `oauth2_interceptor` configuration `pre_auth - @aeneasr you are totally right! It makes more sense.